### PR TITLE
METRON-455: Create stellar management functions to put, get and list data in HDFS

### PR DESCRIPTION
**Pull request by cestella**
Create functions around reading, writing and removing local and HDFS files.

Functions added:
* Interacting with HDFS
  * `HDFS_LS`
  * `HDFS_RM`
  * `HDFS_READ`
  * `HDFS_WRITE`
  * `HDFS_READ_LINES`
* Interacting with the local filesystem
  * `LOCAL_LS`
  * `LOCAL_RM`
  * `LOCAL_READ`
  * `LOCAL_WRITE`
  * `LOCAL_READ_LINES`

You can test these by deploying the management jar as per the instructions in `metron-management` and try them out in the Stellar REPL.

### CODE REVIEW

File: metron-platform/metron-management/README.md [hunk 1]
```diff
@@ -19,11 +19,72 @@ project.
...
+* File functions - Functions around interacting with local or HDFS files
 * Configuration functions - Functions surrounding pulling and pushing configs from zookeeper
 * Parser functions - Functions surrounding adding, viewing, and removing Parser functions.
 * Enrichment functions - Functions surrounding adding, viewing and removing Stellar enrichments as well as managing batch size and index names for the enrichment topology configuration
 * Threat Triage functions - Functions surrounding adding, viewing and removing threat triage functions.
 
+### File Functions
+
+* Local Files
+  * `FILE_LS`
```
Comment (mmiklavc) [inline metron-platform/metron-management/README.md]: Regarding all of the HDFS/FILE functions: How about we change wording a bit? 
FILE_GET/PUT -> LOCAL_READ/WRITE
HDFS_GET/PUT -> HDFS_READ/WRITE

Rationale - make explicit the FS that the reads/writes are using. I had considered simply passing an arg to a unified function (i.e., making FILE_READ/WRITE handle local and hdfs), but then you lose autocomplete. The existing division of labor is more explicit. Also, the GET/PUT names have some terminology overlap with the hdfs shell commands that have different results, e.g. hdfs dfs -get/put operate on files, not file contents. READ/WRITE helps keep a distinction.

File: metron-platform/metron-management/src/main/java/org/apache/metron/management/FileSystemFunctions.java [hunk 2]
```diff
@@ -0,0 +1,402 @@
...
+    };
+
+    FileSystemLs(FileSystemGetter getter) {
+      super(getter);
+    }
+
+    @Override
+    public Object apply(List<Object> args, Context context) throws ParseException {
+
+      String path = (String) args.get(0);
```
Comment (mmiklavc) [inline metron-platform/metron-management/src/main/java/org/apache/metron/management/FileSystemFunctions.java]: Might a sensible default be better here when no args are passed? Perhaps the user's home dir or '/'?
```
[Stellar]>>> HDFS_LS()
[!] Unable to execute: Index: 0, Size: 0
org.apache.metron.common.dsl.ParseException: Unable to execute: Index: 0, Size: 0
...
Caused by: java.lang.IndexOutOfBoundsException: Index: 0, Size: 0
       	at java.util.LinkedList.checkElementIndex(LinkedList.java:555)
       	at java.util.LinkedList.get(LinkedList.java:476)
       	at org.apache.metron.management.FileSystemFunctions$FileSystemLs.apply(FileSystemFunctions.java:213)
```

Comment (cestella) [inline metron-platform/metron-management/src/main/java/org/apache/metron/management/FileSystemFunctions.java]: agreed, homedir makes sense

**Comment by mmiklavc** [comment]
This is excellent... +1 pending the changes pass Travis

**Pull request closed without merging**